app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,40 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def show
+    @market = Market.find(params[:id])
+  end
+
+  def edit
+    @market = Market.find(params[:id])
+  end
+
+  def update
+    @market = Market.find(params[:id])
+    @market.update(market_params)
+    redirect_to market_path
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    Market.create(market_params)
+    redirect_to markets_path
+  end
+
```

Comment:
This generally makes sense. As a minor nit-pick. I tend to see the new and create actions higher up in the controller than the edit and update actions because you have to create objects before you can update them.